### ISSUE TITLE
Playback stops some seconds before the end of the track

### ISSUE THREAD

**Issue by Dysprosium-2** [squeezebox-googlemusic/squeezebox-googlemusic#19]
I installed the Google Music plugin on a Raspberry Pi with Raspbian Jessie and LMS 7.9.0. following the instructions in README.md.
When playing a single track or a playlist the playback of the current track stops between approx. 40 and 10 seconds before the end of the song. In case of a playlist playback continues with the next track, which also ends too early, etc.
I have the same results on a second Raspberry Pi with the current version of Max2Play, LMS 7.9 and the Google Music Plugin, after changing HTTP in HTTPS in line 5 of ProtocolHandler.pm (see issue #15).
I have no idea what the reason for this behaviour is.
Thank you in advance for your help.

**Comment by kettenbach-it** [comment]
An issue on this has been opened in the slimserver repo:

https://github.com/Logitech/slimserver/issues/130

**Comment by Dysprosium-2** [comment]
Thanks for the info!

**Comment by unbehagen** [comment]
To fix this problem, I'm testing the theory that Google doesn't like it if we read the file too slowly. I want to force LMS to buffer the whole file in memory.
I'm experimenting with adding the line
`sub bufferThreshold { 100000 }`
close to the end of ProtocolHandler.pm and restarting LMS. As far as I can tell from other Plugins shipped with LMS, this should in theory tell the Server to buffer up to 100000 kByte (100MByte) before starting the playback. Initially, this did not help. 
When I added the parameter -b 100000:100000 to my squeezelite startup scripts, it so far appears to fix this problem. I listened to a couple of songs so far and none were cut off near the end. I believe you also have to set each player to "Proxied Streaming" in audio settings.

**Comment by frelev** [comment]
@unbehagen did you ever find a general solution for this?
I have the same issue with LMS 7.9 Nightly /max2play on a rpi3. 
Since I have the problem with hardware players I don´t know how to set the buffer client side

EDIT: I found proxied streaming server side. Will try that out

**Comment by frelev** [comment]
I have now tried with proxied streaming and I still have the issue with cuts before the end of the track.
Has anybody found a solution for this?

**Comment by davein2it** [comment]
What do I do ditch lms? ditch google music? help find a fix? This is ruining music for me

**Comment by gsi095** [comment]
Is there a solution anyone has found? still getting the problem

**Comment by davein2it** [comment]
I don't believe so - have moved to spotify which has a plugin that works without this issue.

**Comment by kstunger** [comment]
I would recommend checking out the proxy solution mentioned here:
https://github.com/Logitech/slimserver/issues/130#issuecomment-465507833

**Comment by gsi095** [comment]
Thanks kstunger, I will check it out. If it works I can keep google music and keep the wife approval factor.

I think that google music/youtube music has a killer benefit of premium youtube (no ads and ability to download vids offline) that I dont want to lose going to Spotify
